src/libs/API.js

```diff
@@ -589,6 +601,17 @@ function User_SecondaryLogin_Send(parameters) {
     return request(commandName, parameters);
 }
 
+/**
+ * @param {Object} parameters
+ * @param {String} parameters.base64image
+ * @returns {Promise}
+ */
+function User_UploadAvatar(parameters) {
+    const commandName = 'User_UploadAvatar';
+    requireParameters(['base64image'], parameters, commandName);
```

Comment:
Yep, we will need to convert images to base64 strings to use it. My idea for actual implementation was to use `FileReader.readAsDataURL` to convert the uploaded image to base64 before calling `PersonalDetails.setAvatar()`

Reply:
Perhaps it's better to do the conversation in `PersonalDetails.setAvatar()` rather than before it, but I figured that could be handled when we get around to actually implementing the avatar selector.

Reply:
Yeah I mean it would be great if we could just send the image as a file... like we do here...

https://github.com/Expensify/Expensify.cash/blob/e8084f94737f6a2561bf3cd454b412d93359cb93/src/libs/API.js#L480
https://github.com/Expensify/Web-Expensify/blob/e20da03a31ce8bea2c1b584a4dc631981700ae40/api.php#L559

But guessing this would require modifying that API command so it can handle the `$_FILES`. I'll admit this bugs me a little because: 

- becomes another acceptable way to upload a file to the API from the new app 
- might be confusing to have multiple ways to handle uploading files
- adds more responsibility to the client to encode something (which could also be a very large file / possibly inefficient to encode - kinda making this one up haven't benchmarked this)

I _believe_ we are only sending this as base64 since this is what the library we are using in Web-Expensify outputs as. But while it's convenient there it maybe just adds complexity to `Expensify.cash`.

https://github.com/Expensify/Web-Expensify/blob/e20da03a31ce8bea2c1b584a4dc631981700ae40/site/app/modals/settings/avatarCropModal.jsx#L38-L43

Reply:
Hmm, I agree it would be ideal not to have to rely on converting to base64. We might consider making a new API command that takes in an attachment and calls `ReceiptAPI::upload` like we do in `ReportUtils::createHTMLFromAttachment` and returns the s3url. 

If we do want go that route let's remove `PersonalDetails.setAvatar` until that's completed in Web and we can re-add this function then.

Reply:
Yeah something similar to `ReceiptAPI` could work. I don't know if we need to block progress on this though. Maybe we can create a follow up issue to address it.

Reply:
New issue here: https://github.com/Expensify/Expensify/issues/156268. Also added it to the main user settings rollout plan with a note that we don't need to block on this issue being resolved.